### PR TITLE
test(filterable-select): refactor cypress tests for filterable select FE-4567

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour

Refactor Cypress tests for Filterable Select component to use new cypress-component-react framework for testing.
Add filterable-select.test.js. Also, make minor updates to simple-select.test.js.

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [ ] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `filterable-select.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/select/filterable-select/filterable-select.test.js [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+        commonDataElementInputPreview().type(text);
+        selectInput().should("have.attr", "aria-expanded", "true");
+        selectList().should("be.visible");
+        selectOption(positionOfElement("first"))
+          .should("have.text", optionValue1)
+          .and("be.visible")
+          .and("have.css", "background-color", "rgb(153, 173, 183)");
+        selectOption(positionOfElement("second"))
+          .should("have.text", optionValue2)
+          .and("be.visible");
```
Comment (ZhuoyuJin) [inline src/components/select/filterable-select/filterable-select.test.js:889]: Why there is only background-colour checks only for optionValue 1, not optionValue2 or 3?

**Review by ZhuoyuJin — COMMENTED** [review]
I think "adaptiveLabelBreakpoint" is a prop for filterable select which I don't think has been covered in this test, wondering if there's a way to cover this prop?

**Review by DlgSHi — CHANGES_REQUESTED** [review]
We need to add:
1. `tableHeader` for `FilterableSelect`
and
2. `borderColor` and `fill` for `selectListOption`

File: src/components/select/filterable-select/filterable-select.test.js [hunk 2]
```diff
@@ -0,0 +1,1291 @@
...
+        });
+    });
+
+    it("should call onListScrollBottom event when the list is scrolled to the bottom", () => {
+      CypressMountWithProviders(
+        <FilterableSelectComponent onListScrollBottom={callback} />
+      );
+
+      dropdownButton().click();
+      selectList().scrollTo("bottom").wait(250);
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.test.js:1282]: It didn't scroll right to the bottom when it only scrolled once. Not sure why!

File: src/components/select/filterable-select/filterable-select.test.js (hunk 1 above)
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.test.js:889]: I only checked that the first option had the background change applied as all the other options listed apart from the first one are as they are in a normal unfiltered list. I've added checks now, however, to confirm filtering has not altered the background of the other two options in the list.

File: src/components/select/filterable-select/filterable-select.test.js [hunk 3]
```diff
@@ -0,0 +1,1291 @@
...
+      selectOption(positionOfElement(position)).click();
+      getDataElementByValue("input").should(
+        "have.attr",
+        "value",
+        positionValue
+      );
+    });
+
+    it.each([
+      ["top", "0px", "0px", "0px", "0px"],
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/select/filterable-select/filterable-select.test.js:1001]: I asked Maciek and Ed about these when I did the refactor for the Simple Select component and they said the suffix type positions, e.g. top-start, top-end etc, don't really matter in the context of a Select list. They're more relevant for tooltips as the Select list has the same width as the input, hence why I only included top, bottom, left and right for Simple Select. Maciek said they could actually be removed to simplify the prop interface for the Select components. I'll add one more case for the `auto` position with varying space to verify which position the list will be placed but I'll log a ticket to remove the other options for Selects.

**Comment by DlgSHi** [comment]
> > 1. `tableHeader` for `FilterableSelect`
>
> @DlgSHi the table header is tested by default within the `FilterableSelectMultiColumnsComponent` custom component used to test the multicolumns in a list with table header, body and rows. There is an assertion on the number of columns in the header in this test case. Do you think there should be a separate test case for testing the specific content of the header?

yes please, add it as a separate TC

**Review by DlgSHi — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**